dbms/src/Storages/Transaction/Collator.h

```diff
@@ -42,6 +42,25 @@ class ITiDBCollator : public ICollator
         UTF8_BIN = 83,
     };
 
+    // internal wrapped collator types which are effective for `switch case`
+    enum class CollatorType : uint32_t
+    {
+        // bin
+        UTF8MB4_BIN = 0,
+        UTF8_BIN,
+        LATIN1_BIN,
+        ASCII_BIN,
+        // binary
+        BINARY,
+        // ----
+        UTF8_GENERAL_CI,
+        UTF8MB4_GENERAL_CI,
+        UTF8_UNICODE_CI,
+        UTF8MB4_UNICODE_CI,
+        // ----
+        MAX_,
```

Comment:
Set a big enough value for `MAX_`?

Reply:
> Set a big enough value for `MAX_`?

No need. It's a enum type, compiler will set it automatically.